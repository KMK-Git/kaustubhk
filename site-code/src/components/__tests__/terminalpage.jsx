import React from 'react';
import renderer from 'react-test-renderer';
import TerminalPage from '../terminalpage';
import TerminalPrompt from '../terminalprompt';
import TerminalText from '../terminaltext';
import StyledAnchor from '../styledanchor';

const { act } = renderer;
const user = 'kaustubh@kaustubhk.com:';
const location = '~';

jest.mock('@gatsbyjs/reach-router', () => ({
  ...jest.requireActual('@gatsbyjs/reach-router'),
  useLocation: () => ({
    pathname: '/sampleroute',
  }),
}));

jest.mock('gatsby', () => ({
  __esModule: true,
  useStaticQuery: () => ({
    site: {
      siteMetadata: {
        title: 'Site Title',
        description: 'Site Description',
        siteUrl: 'Site Url',
        image: 'Site Image',
      },
    },
  }),
  graphql: () => ({
    mdx: {
      frontmatter: {
        title: 'Title',
        summary: 'Summary',
        slug: '/slug',
        previewImage: {
          childImageSharp: {
            original: { src: 'image.png', width: 200, height: 200 },
          },
        },
      },
      body: 'var _excluded = ["components"];\n\nfunction _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsxRuntime classic */\n\n/* @jsx mdx */\nvar _frontmatter = {\n  "slug": "/slug",\n  "date": "2022-02-20",\n  "title": "Testing",\n  "summary": "Summary",\n  "priority": 1,\n  "previewImage": "../images/medium/preview4.png",\n  "cardColors": "#fff740,#7afcff,#ff7eb9",\n  "darkCardColors": "#b3aa00,#00b0b3,#b30050"\n};\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = "wrapper";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, _excluded);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: "MDXLayout"\n  }), mdx("p", null, "Testing"));\n}\n;\nMDXContent.isMDXComponent = true;',
    },
    site: { siteMetadata: { siteUrl: 'http://localhost' } },
  }),
}));

describe('TerminalPage', () => {
  it('renders correctly', async () => {
    let tree;
    act(() => {
      tree = renderer
        .create(
          <TerminalPage toolbarText="kaustubh@kaustubhk.com:~">
            <TerminalText isFolder={false}>Hi, Welcome to my website!</TerminalText>
            <br />
            <TerminalPrompt user={user} location={location} text="cat aboutme.txt" />
            <TerminalText isFolder={false}>
              Hi, I&#39;m Kaustubh Khavnekar, a Senior Platform Engineer working for Quantiphi.
              I work primarily on developing architectures capable of solving a given use case.
            </TerminalText>
            <TerminalText isFolder={false}>
              You can read my blogs at&nbsp;
              <StyledAnchor href="https://blogs.kaustubhk.com" target="_blank" rel="noopener noreferrer">https://blogs.kaustubhk.com</StyledAnchor>
            </TerminalText>
            <TerminalPrompt user={user} location={location} text="ls skills/" />
            <TerminalText isFolder>
              &#39;Design Applications&#39;
              &#39;Secure Applications&#39;
              &#39;Develop Applications&#39;
              &#39;Save Money&#39;
              &#39;Develop Automation&#39;
              &#39;Cloud Native&#39;
            </TerminalText>
            <TerminalPrompt user={user} location={location} text="ls -1 certifications/" />
            <TerminalText isFolder={false}>
              <StyledAnchor href="https://www.credly.com/badges/785576dd-bce0-4987-a631-3a2d2b74499c/public_url" target="_blank" rel="noopener noreferrer">
                AWS Certified DevOps Engineer - Professional
              </StyledAnchor>
            </TerminalText>
            <TerminalText isFolder={false}>
              <StyledAnchor href="https://www.credly.com/badges/03b7e0d8-3638-4294-9216-74442fc511cb/public_url" target="_blank" rel="noopener noreferrer">
                CKAD: Certified Kubernetes Application Developer
              </StyledAnchor>
            </TerminalText>
            <TerminalText isFolder={false}>
              <StyledAnchor href="https://www.credly.com/badges/d5c02aa2-a3ef-47d3-93cf-fa6f3753a992/public_url" target="_blank" rel="noopener noreferrer">
                HashiCorp Certified: Terrafom Associate
              </StyledAnchor>
            </TerminalText>
            <TerminalText isFolder={false}>
              <StyledAnchor href="https://www.credly.com/badges/210938c4-48c2-47f0-817c-8d784f07c099/public_url" target="_blank" rel="noopener noreferrer">
                AWS Certified Solutions Architect - Associate
              </StyledAnchor>
            </TerminalText>
            <TerminalText isFolder={false}>
              <StyledAnchor href="https://www.credly.com/badges/5ab7b3ff-3e67-489a-b04e-b864f898ac0e/public_url" target="_blank" rel="noopener noreferrer">
                AWS Certified Developer - Associate
              </StyledAnchor>
            </TerminalText>
            <TerminalPrompt user={user} location={location} text="cat design-credits.txt" />
            <TerminalText isFolder={false}><StyledAnchor href="https://codepen.io/codypearce/pen/JmOQXB" target="_blank" rel="noopener noreferrer">https://codepen.io/codypearce/pen/JmOQXB</StyledAnchor></TerminalText>
          </TerminalPage>,
        );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
