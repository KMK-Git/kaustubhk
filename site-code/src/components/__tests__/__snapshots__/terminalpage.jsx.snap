// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerminalPage renders correctly 1`] = `
[
  <div
    style={
      {
        "background": "linear-gradient(45deg, #57003f 0%,#f57453 100%)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "height": "100%",
        "left": 0,
        "position": "fixed",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      }
    }
  />,
  <div
    style={
      {
        "fontFamily": "Ubuntu",
        "height": "calc(20vh + 400px)",
        "width": "calc(10vw + 600px)",
      }
    }
  >
    <div
      style={
        {
          "boxShadow": "2px 4px 10px rgba(0,0,0,.5)",
          "height": "400px",
          "marginBottom": "10vh",
          "marginLeft": "5vw",
          "marginRight": "5vw",
          "marginTop": "10vh",
          "width": "600px",
        }
      }
    >
      <div
        style={
          {
            "alignItems": "center",
            "background": "linear-gradient(#504b45 0%,#3c3b37 100%)",
            "borderTopLeftRadius": "6px",
            "borderTopRightRadius": "6px",
            "boxSizing": "border-box",
            "display": "flex",
            "height": "25px",
            "padding": "0 8px",
            "width": "100%",
          }
        }
      >
        <div
          style={
            {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-uzwyma{width:12px;height:12px;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;border-radius:100%;padding:0;font-size:7px;text-shadow:0px 1px 0px rgba(255,255,255,0.2);box-shadow:0px 0px 1px 0px #41403A,0px 1px 1px 0px #474642;border:none;margin-right:4px;background:#f25d2b;background:linear-gradient(#f37458 0%, #de4c12 100%);-webkit-background-clip:padding-box;background-clip:padding-box;}.css-uzwyma:hover{cursor:pointer;}.css-uzwyma:focus{outline:none;}",
              }
            }
            data-emotion="css uzwyma"
          />
          <button
            className="css-uzwyma"
          >
            ✕
          </button>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-xpuijx{width:12px;height:12px;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;border-radius:100%;padding:0;font-size:7px;background:linear-gradient(#7d7871 0%, #595953 100%);text-shadow:0px 1px 0px rgba(255,255,255,0.2);box-shadow:0px 0px 1px 0px #41403A,0px 1px 1px 0px #474642;border:none;margin-right:4px;}.css-xpuijx:hover{cursor:pointer;}.css-xpuijx:focus{outline:none;}",
              }
            }
            data-emotion="css xpuijx"
          />
          <button
            className="css-xpuijx"
          >
            ─
          </button>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-xpuijx{width:12px;height:12px;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;border-radius:100%;padding:0;font-size:7px;background:linear-gradient(#7d7871 0%, #595953 100%);text-shadow:0px 1px 0px rgba(255,255,255,0.2);box-shadow:0px 0px 1px 0px #41403A,0px 1px 1px 0px #474642;border:none;margin-right:4px;}.css-xpuijx:hover{cursor:pointer;}.css-xpuijx:focus{outline:none;}",
              }
            }
            data-emotion="css xpuijx"
          />
          <button
            className="css-xpuijx"
          >
            ◻
          </button>
        </div>
        <p
          style={
            {
              "color": "#d5d0ce",
              "fontSize": "12px",
              "lineHeight": "14px",
              "marginLeft": "4px",
            }
          }
        >
          kaustubh@kaustubhk.com:~
        </p>
      </div>
      <div
        style={
          {
            "background": "rgba(56, 4, 40, .9)",
            "fontFamily": "Ubuntu mono",
            "marginTop": "-1px",
            "minHeight": "375px",
            "paddingTop": "2px",
          }
        }
      >
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          Hi, Welcome to my website!
        </div>
        <br />
        <span
          style={
            {
              "display": "inline-block",
              "marginTop": "2px",
            }
          }
        >
          <div
            style={
              {
                "color": "#87d441",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            kaustubh@kaustubhk.com:
          </div>
          <div
            style={
              {
                "color": "#6d85a9",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            ~
          </div>
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            $
          </div>
           
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            cat aboutme.txt
          </div>
        </span>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          Hi, I'm Kaustubh Khavnekar, a Senior Platform Engineer working for Quantiphi. I work primarily on developing architectures capable of solving a given use case.
        </div>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          You can read my blogs at 
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-79mumg{color:#ffffff;}",
              }
            }
            data-emotion="css 79mumg"
          />
          <a
            className="css-79mumg"
            href="https://blogs.kaustubhk.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://blogs.kaustubhk.com
          </a>
        </div>
        <span
          style={
            {
              "display": "inline-block",
              "marginTop": "2px",
            }
          }
        >
          <div
            style={
              {
                "color": "#87d441",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            kaustubh@kaustubhk.com:
          </div>
          <div
            style={
              {
                "color": "#6d85a9",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            ~
          </div>
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            $
          </div>
           
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            ls skills/
          </div>
        </span>
        <div
          style={
            {
              "color": "#006fb8",
            }
          }
        >
          'Design Applications' 'Secure Applications' 'Develop Applications' 'Save Money' 'Develop Automation' 'Cloud Native'
        </div>
        <span
          style={
            {
              "display": "inline-block",
              "marginTop": "2px",
            }
          }
        >
          <div
            style={
              {
                "color": "#87d441",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            kaustubh@kaustubhk.com:
          </div>
          <div
            style={
              {
                "color": "#6d85a9",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            ~
          </div>
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            $
          </div>
           
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            ls -1 certifications/
          </div>
        </span>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-79mumg{color:#ffffff;}",
              }
            }
            data-emotion="css 79mumg"
          />
          <a
            className="css-79mumg"
            href="https://www.credly.com/badges/785576dd-bce0-4987-a631-3a2d2b74499c/public_url"
            rel="noopener noreferrer"
            target="_blank"
          >
            AWS Certified DevOps Engineer - Professional
          </a>
        </div>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-79mumg{color:#ffffff;}",
              }
            }
            data-emotion="css 79mumg"
          />
          <a
            className="css-79mumg"
            href="https://www.credly.com/badges/03b7e0d8-3638-4294-9216-74442fc511cb/public_url"
            rel="noopener noreferrer"
            target="_blank"
          >
            CKAD: Certified Kubernetes Application Developer
          </a>
        </div>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-79mumg{color:#ffffff;}",
              }
            }
            data-emotion="css 79mumg"
          />
          <a
            className="css-79mumg"
            href="https://www.credly.com/badges/d5c02aa2-a3ef-47d3-93cf-fa6f3753a992/public_url"
            rel="noopener noreferrer"
            target="_blank"
          >
            HashiCorp Certified: Terrafom Associate
          </a>
        </div>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-79mumg{color:#ffffff;}",
              }
            }
            data-emotion="css 79mumg"
          />
          <a
            className="css-79mumg"
            href="https://www.credly.com/badges/210938c4-48c2-47f0-817c-8d784f07c099/public_url"
            rel="noopener noreferrer"
            target="_blank"
          >
            AWS Certified Solutions Architect - Associate
          </a>
        </div>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-79mumg{color:#ffffff;}",
              }
            }
            data-emotion="css 79mumg"
          />
          <a
            className="css-79mumg"
            href="https://www.credly.com/badges/5ab7b3ff-3e67-489a-b04e-b864f898ac0e/public_url"
            rel="noopener noreferrer"
            target="_blank"
          >
            AWS Certified Developer - Associate
          </a>
        </div>
        <span
          style={
            {
              "display": "inline-block",
              "marginTop": "2px",
            }
          }
        >
          <div
            style={
              {
                "color": "#87d441",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            kaustubh@kaustubhk.com:
          </div>
          <div
            style={
              {
                "color": "#6d85a9",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            ~
          </div>
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            $
          </div>
           
          <div
            style={
              {
                "color": "#dddddd",
                "display": "inline-block",
                "whiteSpace": "normal",
                "wordBreak": "break-all",
                "wordWrap": "break-word",
              }
            }
          >
            cat design-credits.txt
          </div>
        </span>
        <div
          style={
            {
              "color": "#dddddd",
              "display": "inline-block",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-79mumg{color:#ffffff;}",
              }
            }
            data-emotion="css 79mumg"
          />
          <a
            className="css-79mumg"
            href="https://codepen.io/codypearce/pen/JmOQXB"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://codepen.io/codypearce/pen/JmOQXB
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
